go/v1/resource_descriptor_test.go

```diff
@@ -65,3 +81,23 @@ func TestBadResourceDescriptor(t *testing.T) {
 	err = got.Validate()
 	assert.ErrorIs(t, err, ErrRDRequiredField, "created malformed ResourceDescriptor")
 }
+
+func TestBadResourceDescriptorDigestEncoding(t *testing.T) {
+	got := &ResourceDescriptor{}
+	err := protojson.Unmarshal([]byte(badRdDigestEncoding), got)
+
+	assert.NoError(t, err, "Error during JSON unmarshalling")
+
+	err = got.Validate()
+	assert.ErrorIs(t, err, ErrInvalidDigestEncoding, "created ResourceDescriptor with invalid digest encoding")
```

Comment:
nit/optional: should the message, if this check fails be something like "Did not get expected error when using invalid digest encoding"?

(Similar for below)

Reply:
done.